src/Compilers/CSharp/Portable/Syntax/SyntaxKindFacts.cs

```diff
@@ -1048,7 +1048,7 @@ public static SyntaxKind GetPreprocessorKeywordKind(string text)
 
         public static IEnumerable<SyntaxKind> GetContextualKeywordKinds()
         {
-            for (int i = (int)SyntaxKind.YieldKeyword; i <= (int)SyntaxKind.WhenKeyword; i++)
+            for (int i = (int)SyntaxKind.YieldKeyword; i < (int)SyntaxKind.ElifKeyword ; i++)
```

Comment:
can you add a comment to SyntaKind.ElifKeyword indicating this coupling?  That way no one adds a SyntaxKind before it, and ends up making that SyntaxKind a contextual keyword uninentionally.